### PR TITLE
Refactored the way Play access config

### PR DESCRIPTION
**Pull request by jroper**
* Introduced PlayConfig helper which is more suited to idiomatic use of Typesafe config
* Migrated a number of config options from old paths to new paths
* Removed "modules" from configuration keys
* Moved more settings to reference.conf
* Introduced idiom of using "null" rather than blank or no setting for default values.

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/Configuration.scala [hunk 1]
```diff
@@ -840,3 +847,216 @@ case class Configuration(underlying: Config) {
...
+   *
+   * If the value of the item is null, this will return None, otherwise returns Some.
+   */
+  def getOptional[A: ConfigLoader](path: String): Option[A] = {
+    try {
+      // If the value is null, typesafe config will throw an exception. This is the only way that typesafe config allows
+      // discovering null values
+      Some(get[A](path))
+    } catch {
+      case e: ConfigException.Null => None
```
Comment (richdougherty) [inline framework/src/play/src/main/scala/play/api/Configuration.scala]: Maybe mention that `ConfigException.Missing` will still be thrown for entries that don't have a `null` value? I found it a bit hard to understand until I read the Typesafe Config docs and looked at `PlayConfigSpec`.

**Comment by richdougherty** [comment]
@jroper, I had a minor suggestion for an extra comment but I'm merging this PR now because I know it's blocking you.

**Comment by richdougherty** [comment]
:+1: Overall, a nice improvement.

File: framework/src/play/src/main/scala/play/api/Configuration.scala (hunk 1 above)
Comment (jroper) [inline framework/src/play/src/main/scala/play/api/Configuration.scala]: Will do in another pull request that I'm currently working on.

**Comment by hepin1989** [comment]
for the module configuration,how about ,using the annotation,like
@ModuleConfigPath(path = "play.ws") for ws,then there is no need to using Configuration.getConfig("play.ws") in the WSModule.and.
